base/abstractarray.jl

```diff
@@ -97,36 +99,36 @@ linearindexing(::LinearIndexing, ::LinearIndexing) = LinearSlow()
 end
 
 # check along a single dimension
-checkbounds(::Type{Bool}, sz::Integer, i) = throw(ArgumentError("unable to check bounds for indices of type $(typeof(i))"))
-checkbounds(::Type{Bool}, sz::Integer, i::Real) = 1 <= i <= sz
-checkbounds(::Type{Bool}, sz::Integer, ::Colon) = true
-function checkbounds(::Type{Bool}, sz::Integer, r::Range)
+checkbounds(::Type{Bool}, inds::UnitRange, i) = throw(ArgumentError("unable to check bounds for indices of type $(typeof(i))"))
```

Comment:
not a fan of having an argument with the same name as a closely-related function